### PR TITLE
feat(DataGrid/cell-renderer): PlaygroundCellEditor

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
Custom cell editor is not shared between projects.

**What is the chosen solution to this problem?**
Expose it here.
- `PlaygroundCellEditor` is the editor specific to the playground (it requires methods to fetch semantic types as editor parameters)
- It relies on `RichCellEditor` which allows to dynamically switch between an inline textarea editor and an editor with a dropdown (Datalist)

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/datagrid/src/components/EditableCellRenderer/EditableCellRenderer.component.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	return (
+		<RichCellEditor
+			initialValue={value.value}
+			onChange={setState}
+			// onInit={semanticType ? () => getSemanticType(semanticType) : null}
+			// onFilter={semanticType ? search => getSemanticTypeSuggestions(semanticType, search) : null}
+			// onSubmit={(newValue, updateAllCellsWithValue) => {
+			// 	stopEditing();
+			// 	editCell(data, colDef, newValue, updateAllCellsWithValue);
+			// }}
```
Comment (Benrajalu) [inline packages/datagrid/src/components/EditableCellRenderer/EditableCellRenderer.component.tsx:20]:
```suggestion

```

File: packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+}
+
+function RichCellEditor(props: RichCellEditorPropTypes) {
+	const { onChange, onFilter, onCancel, initialValue, hasSuggestions, isLoading } = props;
+	const [value, setValue] = useState(initialValue);
+
+	const handleInputChange = (
+		_: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+		newValue: CellValue,
+	): void => {
```
Comment (Benrajalu) [inline packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.tsx:30]:
```suggestion
	) => {
```
Not sure we need to set the return type, it should be inferred?

File: packages/datagrid/src/components/RichCellEditor/components/CellEditorDatalist.component.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+			.finally(() => {
+				setIsLoading(false);
+			});
+	};
+	const debouncedHandleUserInput = useCallback(debounce(handleUserInput, 300), []);
+
+	return (
+		<Datalist
+			// Using autoFocus because <Datalist /> does not forward ref, so no manual focus possible
+			autoFocus // eslint-disable-line jsx-a11y/no-autofocus
```
Comment (Benrajalu) [inline packages/datagrid/src/components/RichCellEditor/components/CellEditorDatalist.component.tsx:35]:
```suggestion
			autoFocus // eslint-disable-line jsx-a11y/no-autofocus
```
😬 That's hardcore. Any Datalist rendering would steal focus away and mess up natural tab navigation, right?

Comment (sgangnant) [reply packages/datagrid/src/components/RichCellEditor/components/CellEditorDatalist.component.tsx]: "autoFocus" is not pretty, I agree but I had no other solution as-is (see line above).
This is done to keep the current behavior.

File: packages/datagrid/src/components/PlaygroundCellEditor/PlaygroundCellEditor.component.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	const [state, setState] = useState(value.value);
+	const [isLoading, setIsLoading] = useState(false);
+	const [showApplyTo, setShowApplyToState] = useState(false);
+	const showApplyToRef = useRef(showApplyTo);
+	const applyToRef = useRef<HTMLDivElement>(null);
+	const [applyToStyles, setApplyToStyles] = useState({});
+
+	const [semanticType, setSemanticType] = useState<SemanticType>();
+
+	useImperativeHandle(ref, (): any => ({ getValue: () => state }));
```
Comment (sgangnant) [inline packages/datagrid/src/components/PlaygroundCellEditor/PlaygroundCellEditor.component.tsx:32]: For ag-datagrid

File: packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.tsx (hunk 2 above)
Comment (sgangnant) [reply packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.tsx]: I'm a bot

File: packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+	const [value, setValue] = useState(initialValue);
+
+	const previousCellColor = useRef<string>();
+	React.useEffect(() => {
+		previousCellColor.current = eGridCell.style.color;
+		eGridCell.style.color = 'transparent';
+		return () => {
+			eGridCell.style.color = previousCellColor.current!;
+		};
+	}, []); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (sgangnant) [inline packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.tsx:36]: Ugly solution. This is used to hide the cell content when the cell editor is opened but when the content editor is not yet shown (while loading semantic types for example).

File: packages/datagrid/src/components/RichCellEditor/components/CellEditorDatalist.component.scss [hunk 6]
```diff
@@ -0,0 +1,20 @@
+@use '~@talend/design-tokens/lib/tokens' as tokens;
+
+.cell-editor-datalist {
+	font-family: 'Inconsolata'; // tokens.$coral-data-m; Token not yet available
```
Comment (sgangnant) [inline packages/datagrid/src/components/RichCellEditor/components/CellEditorDatalist.component.scss:4]: :(

File: packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.scss [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+	min-width: 30rem;
+	box-shadow: tokens.$coral-elevation-shadow-neutral-m;
+	border-radius: tokens.$coral-radius-s;
+	padding: tokens.$coral-spacing-m tokens.$coral-spacing-s tokens.$coral-spacing-s;
+	background: tokens.$coral-color-neutral-background;
+
+	display: flex;
+	flex-direction: column;
+	row-gap: tokens.$coral-spacing-m;
+	z-index: 9999999;
```
Comment (Benrajalu) [inline packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.scss:13]:
```suggestion
	z-index: 1;
```
I jest. But also, do I?

File: packages/datagrid/src/components/RichCellEditor/components/CellEditorTextarea.component.scss [hunk 8]
```diff
@@ -0,0 +1,8 @@
+@use '~@talend/design-tokens/lib/tokens' as tokens;
+
+textarea.cell-editor-textarea {
+	margin: 0;
+	padding: tokens.$coral-spacing-xs;
+	font-family: 'Inconsolata'; // tokens.$coral-data-m; Token not yet available
```
Comment (Benrajalu) [inline packages/datagrid/src/components/RichCellEditor/components/CellEditorTextarea.component.scss:6]: it is. An update to @latest should help.

File: packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.scss (hunk 7 above)
Comment (sgangnant) [reply packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.scss]: Removed it, no longer needed since editor is now opened in a "popup" from ag grid's standpoint

**Review by Benrajalu — APPROVED** [review]

File: packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.component.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+	ButtonPrimary,
+	ButtonSecondary,
+	Form,
+	StackHorizontal,
+	StackVertical,
+} from '@talend/design-system';
+
+import theme from './ApplyToIdenticalValues.scss';
+
+interface ApplyToIdenticalValuesPropTypes {
```
[the remark this answers is not in the document]
Comment (tlnd-mhuchet) [reply packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.component.tsx:12]: If it can help: https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases

TLDR; interfaces are the way to go by default, and types comes to rescue for specific cases.

File: packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.scss [hunk 10]
```diff
@@ -0,0 +1,13 @@
+@use '~@talend/design-tokens/lib/tokens';
```
Comment (tlnd-mhuchet) [inline packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.scss:1]:
```suggestion
@use '~@talend/design-tokens/lib/tokens' as tokens;
```

Comment (tlnd-mhuchet) [reply packages/datagrid/src/components/PlaygroundCellEditor/ApplyToIdenticalValues.scss:1]: Is it not mandatory to define the `as myVar` ?

File: packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.scss [hunk 11]
```diff
@@ -0,0 +1,22 @@
+@use '~@talend/design-tokens/lib/tokens';
```
Comment (tlnd-mhuchet) [inline packages/datagrid/src/components/RichCellEditor/RichCellEditor.component.scss:1]:
```suggestion
@use '~@talend/design-tokens/lib/tokens' as tokens;
```
?

**Review by tlnd-mhuchet — APPROVED** [review]
